ast/src/parsed/mod.rs

```diff
@@ -158,11 +160,31 @@ pub struct PolynomialName<T> {
 
 #[derive(Debug, PartialEq, Eq, Default, Clone, PartialOrd, Ord)]
 /// A polynomial with an optional namespace
+/// This is different from SymbolPath mainly due to different formatting.
 pub struct NamespacedPolynomialReference {
-    /// The optional namespace, if `None` then this polynomial inherits the next enclosing namespace, if any
-    pub namespace: Option<String>,
-    /// The name of the polynomial / symbol.
-    pub name: String,
+    pub path: SymbolPath,
+}
```

Comment:
Would it make sense to destruct the path into whatever this struct needs?

Reply:
We could just replace all occurrences of NamespacedPolynomialReference by SymbolPath, but that would have created so many changes that I found it easier like this. Also it's a bit of a conceptual difference as well.